### PR TITLE
Fix SignalExternalWorkflow input should be optional

### PR DESCRIPTION
**Pull request by vancexu**
https://github.com/uber/cadence/issues/3056

### CODE REVIEW

File: service/history/decisionChecker.go [hunk 1]
```diff
@@ -456,9 +456,6 @@ func (v *decisionAttrValidator) validateSignalExternalWorkflowExecutionAttribute
 	if attributes.SignalName == nil {
 		return &workflow.BadRequestError{Message: "SignalName is not set on decision."}
 	}
-	if attributes.Input == nil {
```
Comment (venkat1109) [inline service/history/decisionChecker.go]: do we have integration tests that make sure our clients (go/java) don't break during replay when they encounter signal with no input ?

Comment (vancexu) [reply service/history/decisionChecker.go]: Thanks for pointing out, this PR will actually be a breaking change.

Comment (mfateev) [reply service/history/decisionChecker.go]: I don't think it is breaking as it cannot break already running workflows.

Comment (vancexu) [reply service/history/decisionChecker.go]: @venkat1109 I did some manual tests with both go and java client, this change shouldn't break customer.
@mfateev yes you are right, I confused myself previously, it is not a breaking change.

**Pull request merged**